src/datastructure/KeyframeCollection.cpp

```diff
@@ -111,6 +107,23 @@ FrameworkReturnCode KeyframeCollection::suppressKeyframe(const uint32_t id)
 {
 	std::map< uint32_t, SRef<Keyframe>>::iterator keyframeIt = m_keyframes.find(id);
 	if (keyframeIt != m_keyframes.end()) {
+        auto it = m_refKeyframeToKeyframes.find(id);
+        if (it != m_refKeyframeToKeyframes.end()) {
+            for (const auto& kfId : it->second) {
+                auto itKf = m_keyframes.find(kfId);
+                if (itKf == m_keyframes.end()) {
+                    LOG_ERROR("KeyframeCollection::suppressKeyframe - keyframe {} (whose ref keyframe is {}) does not exist.", kfId, id);
```

Comment:
nit: this should never happen, it would mean that code from l.120 to l.125 did not do its job. It's either an assert (but we almost never run in debug mode), or a crash meaning it needs fixing ASAP.

Or, if we want to be less strict, we could also say that this list is going to be erased anyway at line 118. So we could also continue and make a best effort by setting what we can to null. But I like it less, because it let a potential important error slip.